tests/test_check.py

```diff
@@ -45,148 +45,180 @@ def test_str_representation(self):
-        }
-    )
-
-    monkeypatch.setattr(commands, "_find_dists", lambda a: ["dist/dist.tar.gz"])
-    monkeypatch.setattr(
-        package_file,
-        "PackageFile",
-        pretend.stub(from_filename=lambda *a, **kw: package),
-    )
-
-    # used to crash with `AttributeError`
-    output_stream = io.StringIO()
-    assert not check.check(["dist/*"], output_stream=output_stream)
-    assert output_stream.getvalue() == (
-        "Checking dist/dist.tar.gz: PASSED, with warnings\n"
-        "  warning: `long_description_content_type` missing. "
-        "defaulting to `text/x-rst`.\n"
-        "  warning: `long_description` missing.\n"
-    )
-
-
-def test_strict_fails_on_warnings(monkeypatch, capsys):
-    package = pretend.stub(
-        metadata_dictionary=lambda: {
-            "description": None,
-            "description_content_type": None,
-        }
-    )
-
-    monkeypatch.setattr(commands, "_find_dists", lambda a: ["dist/dist.tar.gz"])
-    monkeypatch.setattr(
-        package_file,
-        "PackageFile",
-        pretend.stub(from_filename=lambda *a, **kw: package),
-    )
-
-    # used to crash with `AttributeError`
-    output_stream = io.StringIO()
-    assert check.check(["dist/*"], output_stream=output_stream, strict=True)
-    assert output_stream.getvalue() == (
-        "Checking dist/dist.tar.gz: FAILED, due to warnings\n"
-        "  warning: `long_description_content_type` missing. "
-        "defaulting to `text/x-rst`.\n"
-        "  warning: `long_description` missing.\n"
-    )
-
-
-def test_check_failing_distribution(monkeypatch):
-    renderer = pretend.stub(render=pretend.call_recorder(lambda *a, **kw: None))
-    package = pretend.stub(
-        metadata_dictionary=lambda: {
-            "description": "blah",
-            "description_content_type": "text/markdown",
-        }
-    )
-    output_stream = io.StringIO()
-    warning_stream = "WARNING"
-
-    monkeypatch.setattr(check, "_RENDERERS", {None: renderer})
-    monkeypatch.setattr(commands, "_find_dists", lambda a: ["dist/dist.tar.gz"])
-    monkeypatch.setattr(
-        package_file,
-        "PackageFile",
-        pretend.stub(from_filename=lambda *a, **kw: package),
-    )
-    monkeypatch.setattr(check, "_WarningStream", lambda: warning_stream)
-
-    assert check.check(["dist/*"], output_stream=output_stream)
-    assert output_stream.getvalue() == (
-        "Checking dist/dist.tar.gz: FAILED\n"
-        "  `long_description` has syntax errors in markup and would not be "
-        "rendered on PyPI.\n"
-        "    WARNING"
-    )
-    assert renderer.render.calls == [pretend.call("blah", stream=warning_stream)]
-
-
-def test_main(monkeypatch):
-    check_result = pretend.stub()
-    check_stub = pretend.call_recorder(lambda a, strict=False: check_result)
-    monkeypatch.setattr(check, "check", check_stub)
-
-    assert check.main(["dist/*"]) == check_result
-    assert check_stub.calls == [pretend.call(["dist/*"], strict=False)]
```

Comment:
I appreciate the instinct to refactor, but it makes for a big diff that's hard to review. I probably would have just renamed the existing test functions to include the word `description`.

Reply:
Indeed, I only realized that when viewing the diff myself :-(

The tests have been slightly modified as well, since they used to be strongly tied to the "long description checking" logic. Now some tests focus on the long description checker, and some tests focus on the infrastructure, and are checker-agnostic.